source/tests/system/nirfmxbt_driver_api_tests.cpp

```diff
@@ -211,6 +258,48 @@ TEST_F(NiRFmxBTDriverApiTests, TxpBasicFromExample_DataLooksReasonable)
   EXPECT_GT(fetched_powers_response.peak_to_average_power_ratio_maximum(), 0.0);
 }
 
+// Note: there aren't any complex attributes in attributes.py, but this at least exercises the code.
+TEST_F(NiRFmxBTDriverApiTests, SetAttributeComplex_ExpectedError)
+{
+  const auto session = init_session(stub(), kPxi5663e);
+
+  EXPECT_RFMX_ERROR(
+      -380231, "This attribute is read-only and cannot be written", session,
+      client::set_attribute_ni_complex_double_array(stub(), session, "", NiRFmxBTAttribute::NIRFMXBT_ATTRIBUTE_ACP_RESULTS_REFERENCE_CHANNEL_POWER, complex_number_array({1.2, 2.2}, {1e6, 1.01e6})));
+}
+
+// Note: there aren't any i8 attributes in attributes.py, but this at least exercises the code.
+TEST_F(NiRFmxBTDriverApiTests, SetAttributeInt8_ExpectedError)
```

Comment:
RFmxBT *does not* have any methods with narrow numerics other that the attribute accesors. So you don't need any other tests for these.  I don't think that we need to add these "identical attribute accessors that do nothing; do nothing" tests for every personality. But it's fine to add these. Also double-check that the service implementation for the setters is doing the same coercion code that specan does.

Reply:
It appears that the Bluetooth service performs the same coercion as the SpecAn service. For example:
[Bluetooth casts to I8](https://github.com/ni/grpc-device/blob/b10041818bfa7f4d4d2ec28e17d83c97568e674c/generated/nirfmxbt/nirfmxbt_service.cpp#L3527) the same way that [SpecAn casts to I8](https://github.com/ni/grpc-device/blob/main/generated/nirfmxspecan/nirfmxspecan_service.cpp#L12172).